File: StringSequence.py
def solution(S):
    count = 0
    new_str = ""

    # checks S first
    if S[0] == S[len(S) - 1]:
        count += 1
        new_str = S[1:] + S[0]

        # if first and last letter in S are similar checks remaining iterations
        for i in range(len(S) - 1):
            if new_str[0] == new_str[len(new_str) - 1]:
                count += 1
                new_str = new_str[1:] + new_str[0]
            else:
                new_str = new_str[1:] + new_str[0]
        print(count) 

    # if first and last letter in S don't match checks the remaining iterations     
    else:
        for i in range(len(S)):
            new_str = S[1:] + S[0]
            
            if new_str[0] == new_str[len(new_str) - 1]:
                count += 1
                new_str = new_str[1:] + new_str[0]
            else:
                new_str = new_str[1:] + new_str[0]
        print(count)
    
solution("abbaa")
solution("abab")
solution("aaaa")

File: StringSequence.js
function solution(S){
    let count = 0
    let new_str = ''
    if (S.charAt(0) === S.charAt(S.length - 1)){
        count += 1
        new_str = S.slice(1) + S.charAt(0)
        for (let i=0; i < (S.length - 1); i++){
            if (new_str.charAt(0) === new_str.charAt(new_str.length - 1)){
                count += 1;
                new_str =  new_str.slice(1) + new_str.charAt(0)
            }else{
                new_str =  new_str.slice(1) + new_str.charAt(0)
            }
        }    
    }
    else{
        new_str = S.slice(1) + S.charAt(0)
        for (let i=0; i < (S.length - 1); i++){

            if (new_str.charAt(0) === new_str.charAt(new_str.length - 1)){
                count += 1;
                new_str =  new_str.slice(1) + new_str.charAt(0)
            }else{
                new_str =  new_str.slice(1) + new_str.charAt(0)
            }
        }
    }
    console.log(count)

}
solution("abbaa")
solution("abab")
solution("aaaa")
solution("aabbab")